Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCiscoFirepower.yaml

```diff
@@ -0,0 +1,324 @@
+                                  )
+          | where ASimMatchingHostname != "No match"
+          | invoke _ASIM_ResolveNetworkProtocol('Protocol')
+          | extend NetworkProtocol = iff(NetworkProtocol == "Unassigned" and Protocol !in (63, 68, 99, 114, 253, 254), Protocol, NetworkProtocol);
+      let Connection_Statistics_Events = AllLogs
+          | where DeviceEventClassID has "RNA:1003:1"
+          | parse-kv AdditionalExtensions as (
+              start: long,
+              end: long,
+              bytesIn: long,
+              bytesOut: long,
+              )
+              with (pair_delimiter=';', kv_delimiter='=') 
+          | lookup EventResultDetailsLookup on Reason
+          | extend
+              SrcBytes = bytesIn,
+              DstBytes = bytesOut,
+              EventOriginalResultDetails = Reason,
+              AdditionalFields = bag_pack("policy", DeviceCustomString1,
+                                                  "instanceID", ProcessID,
+                                                  "clientApplicationID", RequestClientApplication,
+                                                  "clientUrl", RequestURL);
+      let Intrusion_Events = AllLogs
+          | where DeviceEventClassID has "INTRUSION:400"
+          | parse-kv AdditionalExtensions as (
+              start: long
+              )
+              with (pair_delimiter=';', kv_delimiter='=')
+          | extend 
+              EventMessage = Activity,
+              ThreatCategory = DeviceEventCategory,
+              AdditionalFields = bag_pack("policy", DeviceCustomString1,
+                                                  "ipspolicy", DeviceCustomString5,
+                                                  "clientApplicationID", RequestClientApplication,
+                                                  "clientUrl", RequestURL);
+      let Policy_Violation_Events = AllLogs
+          | where DeviceEventClassID has "PV:112"
+          | extend
+              EventMessage = Message,
+              AdditionalFields = bag_pack("policy", DeviceCustomString1)
+          | project-rename DstUsername = DestinationUserName
+          | extend
+              DstUsernameType = _ASIM_GetUsernameType(DstUsername),
+              DstUserType = _ASIM_GetUserType(DstUsername, "");
+      let AllEvents = union Connection_Statistics_Events, Intrusion_Events, Policy_Violation_Events
+          | extend
+              SrcPortNumber = SourcePort,
+              DstPortNumber = DestinationPort,
+              SrcZone = DeviceCustomString3,
+              DstZone = DeviceCustomString4;
+      let Intrusion_Events_ICMP = Intrusion_Events
+          | where NetworkProtocol == "ICMP"
+          | extend
+              NetworkIcmpCode = DestinationPort,
+              NetworkIcmpType = tostring(SourcePort),
+              SrcPortNumber = int(null),
+              DstPortNumber = int(null);
+      union AllEvents, Intrusion_Events_ICMP
```

Comment:
@jayeshprajapaticrest @niralishah-crest 
we are using union heavily here. Please avoid or minimize